docs/user_guide.md

```diff
@@ -402,3 +402,25 @@ you can see the item named "flink-pod-monitor" in the "Service Discovery" sectio
 ### Manage savepoints
 
 See this [doc](./savepoints_guide.md) on how to manage savepoints with the operator.
+
+### Flink Job update
+
+To update a running Flink job's program or execution settings, you can create new savepoint, terminate the job,
+and create a new FlinkCluster with the new program, settings, and savepoint. 
+However, in some cases, you may want to update the Flink job while maintaining the logical continuity
+of the Flink job with the FlinkCluster custom resource. In this case, you can continuously update the spec.job
+of the FlinkCluster custom resource, and the Flink operator takes care of the process required to update the Flink job.
+
+There are several points to note when using the job update feature.
+* To use the job update feature, savepointsDir must be set and the value of this field cannot be deleted when updating.
+This is because the Flink operator requires the value to create a savepoint for job updates.
+* Since fromSavepoint is used only when creating FlinkCluster, it cannot be updated.
```

Comment:
After updating the job, `fromSavepoint` no longer reflects the actual savepoint the job was started from. I am thinking about adding a field in the job status to reflect the actual `fromSavepoint`. With it, we can clearly distinguish that from savepoint1 to savepoint2, it was processed by job v1, from savepoint2 to savepoint3 it was processed by job v2. It might be useful for troubleshooting, say job version2 had a bug which was identified later, then we can rerun the job after fixing the bug from its savepoint2. What do you think?

Reply:
Good idea to add status.fromSavepoint. And for rerun, do you mean to support the Flink job replay in a way that allows updating of spec.fromSavepoint?

Reply:
Support updating `spec.fromSavepoint` would be a nice feature, but it is up to you whether you want to implement it in this PR.

Reply:
I think it is very usable feature, but currently if fromSavepont is allowed to update, new revision will be created always even if only it is updated. What do you think about moving fromSavepont to annotation?

Reply:
I think it would be better to discuss this feature further and add it.

Reply:
What's the problem with "new revision will be created always even if only it is updated"?

Reply:
In more detail, I'm concern about the replay use case using savepoint rather than creating new revision itself. First, I wonder if it is useful for users to record revisions when changing fromSavepoints. If job replay is for data verification or debugging, depending on the user, it may be considered unnecessary to record it as a history. Secondly, I think users may want to replay the same savepoint multiple times for data verification or debugging. In the second case, it cannot be replayed through spec update. In this case, it seems that the replay feature can be supported by annotation. For example like this.

```
annotations:
  user-control: "replay"
```

If you specify a specific savepoint:
```
annotations:
  user-control: "replay: /tmp/savepoint"
```

If it is useful to create a revision for savepoint change, I think that we may support both fromSavepoint field and replay annotation, but I'm not sure if that's okay because it's a rather duplacate feature.

Reply:
To be honest, I don't think creating a new revision for fromSavepoint is a problem. If users really want replaying multiple times, they can create multiple CRs. I feel keeping fromSavepoint in spec is better, making it an annotation only complicates things.

Reply:
Sounds good. Let's proceed like that.

Reply:
Seems we already have `fromSavepoint` in job status. Just need to make sure it is get updated for job update.

https://github.com/GoogleCloudPlatform/flink-on-k8s-operator/blob/master/api/v1beta1/flinkcluster_types.go#L489

Reply:
Enabled `fromSavepoint` updating and introduced replay feature via it.